File: test_data/grammar_samples/sample.lua
-- Lua sample
local function fibonacci(n)
    if n <= 1 then return n end
    return fibonacci(n - 1) + fibonacci(n - 2)
end

local animals = {"cat", "dog", "bird"}

for i, name in ipairs(animals) do
    print(string.format("%d: %s", i, name))
end

local result = fibonacci(10)
print("Fib(10) = " .. tostring(result))

File: test_data/grammar_samples/sample.go
// Go sample
package main

import "fmt"

type Animal struct {
    Name string
    Age  int
}

func (a Animal) Speak() string {
    return fmt.Sprintf("I am %s, age %d", a.Name, a.Age)
}

func main() {
    cat := Animal{Name: "Whiskers", Age: 3}
    fmt.Println(cat.Speak())
}
